src/authorizations/components/redesigned/EditTokenOverlay.tsx

```diff
@@ -9,49 +10,122 @@ import {
+    const {onUpdate, auth} = props
+
+    onUpdate({
+      ...auth,
+      status: auth.status === 'active' ? 'inactive' : 'active',
+    })
   }
 
-  const handleInputChange = event => {
-    setDescription(event.target.value)
+  const onSave = () => {
+    const {onUpdate, auth} = props
+
+    onUpdate({
+      ...auth,
+      description: description,
+    })
+    handleDismiss()
   }
 
   return (
-    <Overlay.Container maxWidth={830}>
+    <Overlay.Container maxWidth={630}>
       <Overlay.Header title="API Token Summary" onDismiss={handleDismiss} />
       <Overlay.Body>
-        <Form>
-          <FlexBox
-            alignItems={AlignItems.Center}
-            direction={FlexDirection.Column}
-            margin={ComponentSize.Large}
-          >
-            <Form.Element label="Description">
-              <Input
-                placeholder="Describe this token"
-                value={description}
-                onChange={handleInputChange}
-                testID="custom-api-token-input"
-              />
-            </Form.Element>
+        <FlexBox
+          alignItems={AlignItems.FlexStart}
+          margin={ComponentSize.Medium}
+          direction={FlexDirection.Column}
+          justifyContent={JustifyContent.SpaceBetween}
+        >
+          <FlexBox margin={ComponentSize.Medium} direction={FlexDirection.Row}>
+            <SlideToggle
+              active={props.auth.status === 'active'}
+              size={ComponentSize.ExtraSmall}
+              onChange={changeToggle}
+            />
+            <InputLabel active={props.auth.status === 'active'}>
+              {labels[props.auth.status]}
+            </InputLabel>
           </FlexBox>
-        </Form>
+          <Form>
+            <FlexBox
+              direction={FlexDirection.Row}
+              margin={ComponentSize.Large}
+              justifyContent={JustifyContent.SpaceBetween}
+            >
+              <Form.Element label="Description">
+                <Input
+                  placeholder="Describe this token"
+                  value={description}
+                  onChange={handleInputChange}
+                  testID="custom-api-token-input"
+                />
+              </Form.Element>
+            </FlexBox>
+            <Page.ControlBarCenter>
+              <FlexBox margin={ComponentSize.Medium}>
+                <Button
+                  color={ComponentColor.Default}
+                  text="Cancel"
+                  onClick={handleDismiss}
+                  testID="token-cancel-btn"
+                />
+                <Button
+                  color={ComponentColor.Primary}
+                  text="Save"
+                  onClick={onSave}
+                  testID="token-save-btn"
+                />
+              </FlexBox>
+            </Page.ControlBarCenter>
+          </Form>
+        </FlexBox>
       </Overlay.Body>
     </Overlay.Container>
   )
 }
+
+const mdtp = {
+  onUpdate: updateAuthorization,
```

Comment:
I was trying to keep the code consistent with other Token components. others have updateAuth set up like line 26 so i figured i hop on that wagon. do you think i should change it?

Reply:
>do you think i should change it?

I do, but since this is already consistent, we can keep it like it is.

I made up a ticket to address it though: https://github.com/influxdata/ui/issues/2511